src/error.rs

```diff
@@ -73,6 +74,17 @@ impl Error {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum SierraAssertError {
+    #[error("casts always happen between numerical types")]
+    Cast,
+    #[error("range should always intersect, from {from_range:?} to {to_range:?}")]
+    Range {
+        from_range: Box<Range>,
+        to_range: Box<Range>,
+    },
```

Comment:
Try to use a single `Box` to keep the error types small:
```suggestion
    #[error("range should always intersect, from {ranges.0:?} to {ranges.1:?}")]
    Range {
        ranges: Box<(Range, Range)>,
    },
```